src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -5322,124 +5258,97 @@ def _set_log_analytics_workspace_extension(cmd, resource_group_name, vm, vm_name
-    disk_encryption_set = DiskEncryptionSet(location=location, tags=tags, identity=encryption_set_identity,
-                                            active_key=key_for_disk_emcryption_set, encryption_type=encryption_type,
-                                            rotation_to_latest_key_version_enabled=enable_auto_key_rotation)
-
-    if federated_client_id is not None:
-        disk_encryption_set.federated_client_id = federated_client_id
-
-    return sdk_no_wait(no_wait, client.begin_create_or_update, resource_group_name, disk_encryption_set_name,
-                       disk_encryption_set)
+def assign_disk_encryption_set_identity(cmd, resource_group_name, disk_encryption_set_name,
+                                        mi_system_assigned=None, mi_user_assigned=None):
+    from .aaz.latest.disk_encryption_set import Update as _Update, Show as _Show
 
+    args = {
+        "disk_encryption_set_name": disk_encryption_set_name,
+        "resource_group": resource_group_name
+    }
+    identity = _Show(cli_ctx=cmd.cli_ctx)(command_args=args).get('identity', {})
+    _user_mi = identity.pop('userAssignedIdentities', None)
+    if _user_mi:
+        identity.update({'user_assigned_identities': _user_mi})
 
-def update_disk_encryption_set(cmd, instance, client, resource_group_name, key_url=None, source_vault=None,
-                               enable_auto_key_rotation=None, federated_client_id=None):
-    from azure.mgmt.core.tools import resource_id, is_valid_resource_id
-    from azure.cli.core.commands.client_factory import get_subscription_id
-    if key_url:
-        instance.active_key.key_url = key_url
+    existing_system_identity = False
+    existing_user_identities = set()
+    if identity:
+        existing_system_identity = identity['type'] in ['SystemAssigned', 'SystemAssigned, UserAssigned']
+        existing_user_identities = {x.lower() for x in list(identity.get('user_assigned_identities', {}).keys())}
 
-    if source_vault:
-        if not is_valid_resource_id(source_vault):
-            source_vault = resource_id(subscription=get_subscription_id(cmd.cli_ctx),
-                                       resource_group=resource_group_name,
-                                       namespace='Microsoft.KeyVault', type='vaults', name=source_vault)
-        instance.active_key.source_vault = {'id': source_vault}
+    add_system_assigned = mi_system_assigned
+    add_user_assigned = {x.lower() for x in (mi_user_assigned or [])}
 
-    if enable_auto_key_rotation is not None:
-        instance.rotation_to_latest_key_version_enabled = enable_auto_key_rotation
+    updated_system_assigned = existing_system_identity or add_system_assigned
+    updated_user_assigned = list(existing_user_identities.union(add_user_assigned))
 
-    if federated_client_id is not None:
-        instance.federated_client_id = federated_client_id
+    identity['type'] = 'SystemAssigned'
```

Comment:
What if only assigning user assigned identity?

Reply:
The type will be `UserAssigned` if only assigned user mi.
https://github.com/cxznmhdcxz/azure-cli/blob/ded7f68b6d0a77f2982215c9943553d1c9bfe1d0/src/azure-cli/azure/cli/command_modules/vm/custom.py#L5286-L5288

Reply:
What if neither user nor system assigned identities being assigned, although it is an edge case.